lib/test/registerUser.js

```diff
@@ -1,54 +1,74 @@
-const BitcoreLib = require('@dashevo/dashcore-lib');
+const {
+  PrivateKey,
+  PublicKey,
+  Transaction,
+} = require('@dashevo/dashcore-lib');
 
-const { PrivateKey, Transaction } = BitcoreLib;
+const IdentityPublicKey = require(
+  '@dashevo/dpp/lib/identity/IdentityPublicKey',
+);
+const IdentityCreateTransition = require(
+  '@dashevo/dpp/lib/identity/stateTransitions/identityCreateTransition/IdentityCreateTransition',
+);
 
 /**
  * Register user
- * @param {string} userName
- * @param {RpcClient} api
- * @returns {Promise<{ userId: string, privateKeyString: string, address: string }>}
+ *
+ * @param {RpcClient} coreAPI
+ * @param {DapiClient} dapiClient
+ *
+ * @returns {Promise<{ identityId: string, identityPrivateKey: PrivateKey }>}
  */
-async function registerUser(userName, api) {
-  let response = await api.getnewaddress();
+async function registerUser(coreAPI, dapiClient) {
+  let response = await coreAPI.getnewaddress();
   const address = response.result;
 
-  response = await api.dumpprivkey(address);
+  response = await coreAPI.dumpprivkey(address);
   const privateKeyString = response.result;
 
   const privateKey = new PrivateKey(privateKeyString);
+  const publicKey = PublicKey.fromPrivateKey(privateKey);
+  // eslint-disable-next-line no-underscore-dangle
+  const publicKeyHash = publicKey._getId();
 
-  await api.generate(101);
-  await api.sendtoaddress(address, 10);
-  await api.generate(7);
+  await coreAPI.generate(101);
+  await coreAPI.sendtoaddress(address, 10);
+  await coreAPI.generate(7);
 
-  response = await api.listunspent();
+  response = await coreAPI.listunspent();
   const unspent = response.result;
   const inputs = unspent.filter(input => input.address === address);
 
-  const transactionPayload = new Transaction.Payload.SubTxRegisterPayload();
+  const transaction = new Transaction();
 
-  transactionPayload.setUserName(userName)
-    .setPubKeyIdFromPrivateKey(privateKey)
+  transaction.from(inputs)
+    .addBurnOutput(10000, publicKeyHash)
     .sign(privateKey);
 
-  const transaction = new Transaction({
-    type: Transaction.TYPES.TRANSACTION_SUBTX_REGISTER,
-    version: 3,
-    extraPayload: transactionPayload.toString(),
-  });
+  await coreAPI.sendrawtransaction(transaction.serialize());
 
-  transaction.from(inputs)
-    .addFundingOutput(10000)
-    .change(address)
-    .fee(668)
-    .sign(privateKey);
+  const outPoint = transaction.getOutPointBuffer(0)
+    .toString('base64');
+
+  const identityPrivateKey = new PrivateKey();
+
+  const identityCreateTransition = new IdentityCreateTransition({
+    lockedOutPoint: outPoint,
+    publicKeys: [
+      {
+        id: 0,
+        type: IdentityPublicKey.TYPES.ECDSA_SECP256K1,
+        data: PublicKey.fromPrivateKey(identityPrivateKey).toString(),
+        isEnabled: true,
+      },
+    ],
+  });
 
-  const { result: userId } = await api.sendrawtransaction(transaction.serialize());
+  await dapiClient.updateState(identityCreateTransition);
```

Comment:
We may need to wait for a little bit for the transaction to be confirmed. (Note that since v14 _all_ transactions are instant, so it shouldn't be longer than 2 secs on average) We do not broadcast isntalocks through a DAPI streaming endpoint, do we?

Reply:
@antouhou I don't think we're broadcasting them. But I'll check.

Reply:
Just add 2-3 sec wait here

Reply:
I've meant add wait before the update state, not after :grin: You need to confirm burn transaction before you've updating the state :grin: